src/vs/workbench/parts/execution/electron-browser/execution.contribution.ts

```diff
@@ -85,20 +85,26 @@ CommandsRegistry.registerCommand({
 		const fileService = accessor.get(IFileService);
 		const integratedTerminalService = accessor.get(IIntegratedTerminalService);
 		const terminalService = accessor.get(ITerminalService);
-		resource = getResourceForCommand(resource, accessor.get(IListService), editorService);
-
-		return fileService.resolveFile(resource).then(stat => {
-			return stat.isDirectory ? stat.resource.fsPath : paths.dirname(stat.resource.fsPath);
-		}).then(directoryToOpen => {
-			if (configurationService.getValue<ITerminalConfiguration>().terminal.explorerKind === 'integrated') {
-				const instance = integratedTerminalService.createInstance({ cwd: directoryToOpen }, true);
-				if (instance) {
-					integratedTerminalService.setActiveInstance(instance);
-					integratedTerminalService.showPanel(true);
+		let resources:uri[] = [];
+		let directoryMap:Map<string, boolean> = new Map();
+		resources = getMultiSelectedResources(resource, accessor.get(IListService), editorService);
+		return resources.map((resource) => {
+			return fileService.resolveFile(resource).then(stat => {
+				return stat.isDirectory ? stat.resource.fsPath : paths.dirname(stat.resource.fsPath);
+			}).then(directoryToOpen => {
+				if (!directoryMap.has(directoryToOpen)) {
+					directoryMap.set(directoryToOpen, true);
+					if (configurationService.getValue<ITerminalConfiguration>().terminal.explorerKind === 'integrated') {
+						const instance = integratedTerminalService.createInstance({ cwd: directoryToOpen }, true);
+						if (instance) {
```

Comment:
Instead of setting active and showing the panel for every instance, how about we set and show only once for the "primary" selection (I think that's a thing @isidorn?).

Reply:
Makes sense.

Reply:
Sure will check.

Reply:
@sriram-dev primary selection is the resource passed in the command. The getResourcesForCommand gets all the other selected elements (including the primary)